ml-proto/spec/kernel.ml

```diff
@@ -119,16 +119,18 @@ and func' =
 type memory = memory' Source.phrase
 and memory' =
 {
-  initial : Memory.size;
+  min : Memory.size;
   max : Memory.size;
   segments : segment list;
 }
 and segment = Memory.segment Source.phrase
 
 type export = export' Source.phrase
 and export' =
-  | ExportFunc of string * var
-  | ExportMemory of string
+{
+  name : string;
+  kind : [`Func of var | `Memory]
```

Comment:
Ah, thanks goodness, much nicer.  I had wanted to name these as fields in [design/#659](https://github.com/WebAssembly/design/pull/659).